src/util/thread_metrics.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs;
+use std::io::{Result, Error, ErrorKind, Read};
+use std::sync::Mutex;
+
+use prometheus::{self, Opts, CounterVec, Desc, proto, Collector};
+
+use libc::{self, pid_t};
+
+/// monitor current process's threads.
+pub fn monitor_threads<S: Into<String>>(namespace: S) -> Result<()> {
+    let pid = unsafe { libc::getpid() };
+    let tc = ThreadsColletcor::new(pid, namespace);
+    try!(prometheus::register(Box::new(tc)).map_err(|e| to_err(format!("{:?}", e))));
+
+    Ok(())
+}
+
+struct ThreadsColletcor {
+    pid: pid_t,
+    descs: Vec<Desc>,
+    cpu_totals: Mutex<CounterVec>,
+}
+
+impl ThreadsColletcor {
+    fn new<S: Into<String>>(pid: pid_t, namespace: S) -> ThreadsColletcor {
+        let cpu_totals = CounterVec::new(Opts::new("thread_cpu_seconds_total",
+                                                   "Total user and system CPU time spent in \
+                                                    seconds by threads.")
+                                             .namespace(namespace),
+                                         &["name", "tid"])
+            .unwrap();
+        let descs = cpu_totals.desc().into_iter().cloned().collect();
+
+        ThreadsColletcor {
+            pid: pid,
+            descs: descs,
+            cpu_totals: Mutex::new(cpu_totals),
+        }
+    }
+}
+
+impl Collector for ThreadsColletcor {
+    fn desc(&self) -> Vec<&Desc> {
+        self.descs.iter().collect()
+    }
+
+    fn collect(&self) -> Vec<proto::MetricFamily> {
+        let cpu_totals = self.cpu_totals.lock().unwrap();
+        let pairs = get_threads(self.pid).unwrap();
+        for (tid, tname) in pairs {
+            if let Ok((utime, stime)) = find_thread_cpu_time(self.pid, tid) {
+                let total = (utime + stime) / *CLK_TCK;
```

Comment:
No, `tick` is the actually unit of those nembers. I think it means:

    tick = second / sysconf(_SC_CLK_TCK)